### PR TITLE
feat(NODE-4014): Add let option to bulk write operations

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR adds support for the `let` option for bulk operations.  This PR also includes a fix to the crud spec tests from this PR: https://github.com/mongodb/specifications/pull/1154.

The failing tests are unrelated to the changes in this PR and are failing on main. 

- adl tests failing are a known issue across drivers
- socks5 & custom csfle tests are failing against master in libmongocrypt - this is a known issue and is being investigated

##### Is there new documentation needed for these changes?

Yes.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

**Comment by baileympearson** [comment]
evergreen rebuild

**Comment by baileympearson** [comment]
evergreen retry

File: test/tools/unified-spec-runner/operations.ts [hunk 1]
```diff
@@ -173,7 +173,10 @@ operations.set('assertNumberConnectionsCheckedOut', async ({ entities, operation
 
 operations.set('bulkWrite', async ({ entities, operation }) => {
   const collection = entities.getEntity('collection', operation.object);
-  return collection.bulkWrite(operation.arguments.requests);
+  const operations = operation.arguments.requests;
```
Comment (dariakp) [inline test/tools/unified-spec-runner/operations.ts:176]: You can do this a little bit more concisely like in some other operations here by doing:
`const { requests: operations, ...options } = operation.arguments;`

**Review by dariakp — CHANGES_REQUESTED** [review]
Just a small suggestion, otherwise LGTM

**Review by dariakp — APPROVED** [review]

**Review by nbbeeken — APPROVED** [review]

**Review by durran — APPROVED** [review]

**Pull request merged**